opm/output/eclipse/summaryComparator.cpp

```diff
@@ -0,0 +1,243 @@
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <opm/output/eclipse/summaryComparator.hpp>
+#include <ert/ecl/ecl_sum.h>
+#include <ert/util/stringlist.h>
+#include <ert/util/int_vector.h>
+#include <ert/util/bool_vector.h>
+#include <opm/common/ErrorMacros.hpp>
+
+bool SummaryComparator::open(const char* basename1, const char* basename2){
+  //opens the files
+  
+  ecl_sum1 = ecl_sum_fread_alloc_case(basename1, ":");
+  ecl_sum2 = ecl_sum_fread_alloc_case(basename2, ":");
+  if (ecl_sum1 == nullptr || ecl_sum2 == nullptr) {
+    OPM_THROW(std::runtime_error, "Not able to open files");
+  }
+  else {
+    return true;
+  }
+}
+
+void SummaryComparator::close() {
+  ecl_sum_free(ecl_sum1);
+  ecl_sum_free(ecl_sum2);
+  
+  stringlist_free(keys1);
+  stringlist_free(keys2);
+  
+}
+
+
+void SummaryComparator::setKeys(){
+  ecl_sum_select_matching_general_var_list( ecl_sum1 , "*" , this->keys1);
+  stringlist_sort(this->keys1 , nullptr );
+  
+  ecl_sum_select_matching_general_var_list( ecl_sum2 , "*" , this->keys2);
+  stringlist_sort(this->keys2 , nullptr );
+
+  if(stringlist_get_size(keys1) <= stringlist_get_size(keys2)){
+    this->keysShort = this->keys1;
+    this->keysLong = this->keys2;
+  }else{
+    this->keysShort = this->keys2;
+    this->keysLong = this->keys1;
+  }
+}
+
+
+
+Deviation SummaryComparator::calculateDeviations(double val1, double val2){
+  //takes in two values as argument, calculates the deviation. returns it as a struct Deviation
+  double absDev, relDev;
+  Deviation deviation; 
+  if(val1 >= 0 && val2 >= 0){
+    absDev =  std::max(val1, val2) - std::min(val1, val2);
+      
+    deviation.abs = absDev;
+    if(val1 == 0 && val2 == 0){}
+    else{
+      relDev = absDev/double(std::max(val1, val2));
+      deviation.rel = relDev;
+    }
+
+  }
+  return deviation;
+}
+
+
+
+
+
+void SummaryComparator::setTimeVecs(std::vector<double> &timeVec1,std::vector<double> &timeVec2){
+ 
+  for (int time_index = 0; time_index < ecl_sum_get_data_length(ecl_sum1); time_index++){
+    timeVec1.push_back(ecl_sum_iget_sim_days(ecl_sum1 , time_index ));
+  }
+  
+
+  for (int time_index = 0; time_index < ecl_sum_get_data_length(ecl_sum2); time_index++){
+    timeVec2.push_back(ecl_sum_iget_sim_days(ecl_sum2 , time_index ));
+  }
+}
+
+
+
+
+//Read the data from the two files into separate vectors. Not necessary the same amount of values, but the values correspond to the same interval in time. Thus possible to interpolate values.
+ 
+void SummaryComparator::getDataVecs(std::vector<double> &dataVec1,std::vector<double> &dataVec2,const char* keyword){  
+  for (int time_index = 0; time_index < ecl_sum_get_data_length(ecl_sum1); time_index++){
+    dataVec1.push_back(ecl_sum_iget(ecl_sum1, time_index, ecl_sum_get_general_var_params_index( ecl_sum1 , keyword )));
```

Comment:
since this vector potentially can have some size i think you should use dataVec1.reserve(ecl_sum_get_data_length(ecl_sum1); to avoid repeated memory reallocation during the pushback() calls.